File: pieceofcake2.py
n,h,v=map(int,input().split())
if(2<=n<=10000 and 0<h<n and 0<v<n):
  if(n-h > h): h=n-h
  if(n-v > v): v=n-v
  print((h*v*4))

File: pieceofcake2.rb
n,h,v=gets.chomp.split.map(&:to_i)
if(n.between?(2,10000) and h.between?(0,n) and v.between?(0,n))
  h=n-h if(n-h > h) 
  v=n-v if(n-v > v) 
  print(h*v*4)
end